crash_rate_aggregates/fill_database.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python
+
+import re
+import sys
+from datetime import datetime, date, timedelta
+import dateutil.parser
+
+import numpy as np
+
+from moztelemetry.spark import get_pings, get_pings_properties
+
+import pyspark.sql.types as types
+
+FRACTION = 1.0
+
+# paths/dimensions within the ping to compare by, in the same format as the second parameter to `get_pings_properties`
+# in https://github.com/mozilla/python_moztelemetry/blob/master/moztelemetry/spark.py
+COMPARABLE_DIMENSIONS = [
+    "environment/build/version",
+    "environment/build/buildId",
+    "application/channel",
+    "application/name",
+    "environment/system/os/name",
+    "environment/system/os/version",
+    "environment/build/architecture",
+    "meta/geoCountry",
+    "environment/addons/activeExperiment/id",
+    "environment/addons/activeExperiment/branch",
+    "environment/settings/e10sEnabled",
+]
+
+# names of the comparable dimensions above, used as dimension names in the database
+DIMENSION_NAMES = [
+    "build_version",
+    "build_date",
+    "channel",
+    "application",
+    "os_name",
+    "os_version",
+    "architecture",
+    "country",
+    "experiment_id",
+    "experiment_branch",
+    "e10s_enabled",
+]
+assert len(COMPARABLE_DIMENSIONS) == len(DIMENSION_NAMES)
+
+INSERT_CHUNK_SIZE = 500 # number of records to accumulate in a single database request; higher values mean faster database insertion at the expense of memory usage
+
+def compare_crashes(pings, start_date, end_date, comparable_dimensions, dimension_names):
+    """Returns a PairRDD where keys are user configurations and values are Numpy arrays of the form [usage hours, main process crashes, content process crashes, plugin crashes]"""
+    ping_properties = get_pings_properties(pings, comparable_dimensions + [
```

Comment:
Accessing the properties directly, i.e. without `get_pings_properties`, might be a lot faster.